### PR TITLE
Fix, update, and enable long running tests

### PR DESCRIPTION
**Pull request by hossam-nasr**
This PR enables long running tests and fixes all the tests to our testing suite so that all our tests, including E2E and long running tests, finally pass :) Java and Ballerina long running tests have been disabled, pending installing Java/Ballerina on the new pipeline images, tracked here: https://github.com/microsoft/vscode-azurefunctions/issues/4210. PR changes include:

- Removing deprecated/obsolete tests, including:
    - C# Script and Functions v1 tests
    - Functions v2 and v3 tests
    - .NET Core 2.1 and 3.1 tests
    - Node 12 and 14
    - Python 3.7
    - Deprecated create new project APIs

- Adding tests for new language versions, including:
   - .NET 8.0
   - PowerShell 7.2 -> PowerShell 7.4
   - Python 3.8, 3.9, 3.10, and 3.11
   - Node 18 and 20

- Removing extra prompts and adding necessary ones, such as removing prompts for subscriptions that aren't needed or adding prompts for language model versions

- Re-enabling Python and C# create and deploy tests on Windows (which is actually the only agent running right now on the new pipelines)

- Removing `package-lock.json` and running `npm install` again to update the `package-lock` file, in order to, among other upgrades, upgrade the `chokidar` package to `3.6.0`, to avoid running into this issue which was causing the TypeScript create and deploy test to fail: https://github.com/paulmillr/chokidar/issues/1299

### CODE REVIEW

File: test/nightly/functionAppOperations.test.ts [hunk 1]
```diff
@@ -42,7 +40,7 @@ suite('Function App Operations', function (this: Mocha.Suite): void {
     });
 
     test('Create - Advanced', async () => {
-        const testInputs: (string | RegExp)[] = [appName, /\.net/i, 'Windows', '$(plus) Create new resource group', rgName, location, 'Consumption', '$(plus) Create new storage account', saName, '$(plus) Create new Application Insights resource', aiName];
+        const testInputs: (string | RegExp)[] = [appName, 'Consumption', location, /\.net/i, 'Windows', '$(plus) Create new resource group', rgName, '$(plus) Create new storage account', saName, '$(plus) Create new Application Insights resource', aiName];
```
Comment (nturinski) [inline test/nightly/functionAppOperations.test.ts:43]: Future note: We should add tests for creating host plans and flex consumption.

File: test/project/createNewProject.test.ts [hunk 2]
```diff
@@ -8,67 +8,64 @@ import { FuncVersion, JavaBuildTool, ProjectLanguage, TemplateSource } from '../
...
 
 interface CreateProjectTestCase extends ICreateProjectTestOptions {
     description?: string;
 }
 
 const testCases: CreateProjectTestCase[] = [
-    { ...getCSharpValidateOptions('netcoreapp2.1', FuncVersion.v2) },
-    { ...getCSharpValidateOptions('netcoreapp3.1', FuncVersion.v3), inputs: [/3/], description: 'netcoreapp3.1' },
     { ...getCSharpValidateOptions('net6.0', FuncVersion.v4), inputs: [/6/], description: 'net6.0' },
     { ...getCSharpValidateOptions('net6.0', FuncVersion.v4), inputs: [/6.*isolated/i], description: 'net6.0 isolated' },
```
Comment (nturinski) [inline test/project/createNewProject.test.ts:19]: I think we can get rid of `net6.0` here, but we should replace it with `.net8.0`

File: test/project/createNewProject.test.ts [hunk 3]
```diff
@@ -8,67 +8,64 @@ import { FuncVersion, JavaBuildTool, ProjectLanguage, TemplateSource } from '../
...
+
+// test python v1 model
+testCases.push({
+    ...getPythonValidateOptions('.venv', FuncVersion.v4),
+    inputs: [/Model V1/i, TestInput.UseDefaultValue]
+});
+
+
+/* temporarily disable Java and Ballerina tests until we can install Java on the new pipelines
+
```
Comment (nturinski) [inline test/project/createNewProject.test.ts:42]: Could you link the GitHub issue you made for this?

File: test/project/validateProject.ts [hunk 4]
```diff
@@ -89,11 +89,11 @@ export function getCSharpValidateOptions(targetFramework: string, version: FuncV
             'azureFunctions.projectLanguage': ProjectLanguage.CSharp,
             'azureFunctions.projectRuntime': version,
             'azureFunctions.preDeployTask': 'publish (functions)',
-            'azureFunctions.deploySubpath': path.join(projectSubpath ?? '', `bin/Release/${targetFramework}/publish`),
+            'azureFunctions.deploySubpath': `${projectSubpath ? `${projectSubpath}/` : ''}bin/Release/${targetFramework}/publish`,
```
Comment (alexweininger) [inline test/project/validateProject.ts:92]: Why does `path.join` no longer work here?

File: test/project/createNewPythonProject.test.ts [hunk 5]
```diff
@@ -25,16 +25,16 @@ suite('Create New Python Project', () => {
         }
         this.timeout(2 * 60 * 1000);
 
-        const alias: string = process.platform === 'win32' ? 'py -3.7' : 'python3.7';
+        const alias: string = process.platform === 'win32' ? 'py -3.9' : 'python3.9';
         await runWithTestActionContext('createProject', async context => {
-            await createAndValidateProject(context, { ...getPythonValidateOptions('.venv', FuncVersion.v2), inputs: [/enter/i, alias] });
+            await createAndValidateProject(context, { ...getPythonValidateOptions('.venv', FuncVersion.v4), inputs: [/Model V1/, /enter/i, alias] });
```
Comment (alexweininger) [inline test/project/createNewPythonProject.test.ts:30]: nit: you could make `/Model V1/` a constant and use it throughout this file.

File: test/project/validateProject.ts (hunk 4 above)
Comment (hossam-nasr) [reply test/project/validateProject.ts:92]: Two reasons: 

For the `deploySubpath`, we actuallly hard-code it to this directory when we initialize the project for use in VSCode. Why that's the case, I'm not sure, but @nturinski's advice was to not mess with it. Even doing something like `path.join('', 'path/to/file');` converts it to `path\\to\\file`.

For the `expectedPaths`, these are used by globby to check for matching files, and apparently globby doesn't support Windows-style path separators

Comment (alexweininger) [reply test/project/validateProject.ts:92]: Ok. Makes sense 👍

File: test/nightly/functionAppOperations.test.ts (hunk 1 above)
Comment (hossam-nasr) [reply test/nightly/functionAppOperations.test.ts:43]: https://github.com/microsoft/vscode-azurefunctions/issues/4214

File: test/project/createNewPythonProject.test.ts (hunk 5 above)
Comment (hossam-nasr) [reply test/project/createNewPythonProject.test.ts]: 🫡

**Review by nturinski — APPROVED** [review]

**Review by alexweininger — APPROVED** [review]

**Pull request merged**